### PR TITLE
machine-config-operator: Update OKD config

### PR DESCRIPTION
**Pull request by LorbusChris**

### CODE REVIEW

File: ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10__okd.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    tag: rhel-8-golang-1.16-openshift-4.9
+build_root:
+  from_repository: true
+images:
+- dockerfile_path: Dockerfile
+  from: base
+  inputs:
+    ocp_builder_rhel-8-golang-1.16-openshift-4.9:
+      as:
+      - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.9
```
Comment (sinnykumari) [inline ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10__okd.yaml:18]: any reason not to use rhel-8-golang-1.16-openshift-4.10?

Comment (LorbusChris) [reply ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10__okd.yaml:18]: yes, that's the one currently still being used for OCP on master: https://github.com/openshift/release/blob/master/ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-master.yaml#L33 (lmk if you'd like me to update that too)

Comment (LorbusChris) [reply ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10__okd.yaml:18]: Actually, the still unused config for release-4.10 has it set correctly: https://github.com/openshift/release/blob/master/ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10.yaml#L33. Should I change just update release-4.10__okd, or also master__okd?

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10-presubmits.yaml [hunk 2]
```diff
@@ -2258,6 +2258,355 @@ presubmits:
...
+    - ^release-4\.10$
+    - ^release-4\.10-
+    cluster: build02
+    context: ci/prow/okd-e2e-aws
+    decorate: true
+    labels:
+      ci-operator.openshift.io/variant: okd
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-machine-config-operator-release-4.10-okd-e2e-aws
```
[the remark this answers is not in the document]
Comment (LorbusChris) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10-presubmits.yaml:2273]: good catch, thank you. Will update

File: ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10__okd.yaml (hunk 1 above)
Comment (LorbusChris) [reply ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10__okd.yaml:18]: Ok, weird. I was sure I saw 4.9 on the first link above, but apparently I was wrong. So, nvmd, I'll update this to 4.10, too.

**Review by kikisdeliveryservice — COMMENTED** [review]
Q: is there a reason we are adding the vsphere job as always_run true? They've been pretty iffy lately:
https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/pull-ci-openshift-machine-config-operator-master-e2e-vsphere

In main repo we _don't_ run them all the time (see: https://github.com/openshift/release/blob/7647e5b981b0a163c9cac2647da004579ad69681/ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10-presubmits.yaml#L1998) so not sure why we'd run them all the time here?

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10-presubmits.yaml [hunk 3]
```diff
@@ -2258,6 +2258,359 @@ presubmits:
...
+              name: cluster-profile-gcp
+      - name: pull-secret
+        secret:
+          secretName: registry-pull-credentials
+      - name: result-aggregator
+        secret:
+          secretName: result-aggregator
+    trigger: (?m)^/test( | .* )okd-e2e-upgrade,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10-presubmits.yaml:2491]:
```suggestion
    always_run: false
```

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.9-presubmits.yaml [hunk 4]
```diff
@@ -2259,6 +2259,359 @@ presubmits:
...
+              name: cluster-profile-gcp
+      - name: pull-secret
+        secret:
+          secretName: registry-pull-credentials
+      - name: result-aggregator
+        secret:
+          secretName: result-aggregator
+    trigger: (?m)^/test( | .* )okd-e2e-upgrade,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.9-presubmits.yaml:2492]:
```suggestion
    always_run: false
```

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10-presubmits.yaml (hunk 3 above)
Comment (LorbusChris) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.10-presubmits.yaml:2491]: Good catch, will update, thank you!

**Comment by kikisdeliveryservice** [comment]
Thanks for the update @LorbusChris 

/lgtm

**Pull request merged**